src/db/pool.cr

```diff
@@ -47,6 +57,49 @@ module DB
       @initial_pool_size.times { build_resource }
     end
 
+    def reap_connection!(connection : T, delay : Float64)
+      sleep delay
+
+      sync do
+        connection_idle = @idle.includes? connection
+        return unless connection_idle # let the next reaping attempt get it
+
+        delete connection
+      end
+
+      connection.health_check! if connection.responds_to?(:health_check!)
+
+      sync do
+        @total << connection
+        @idle << connection
```

Comment:
theoretically, this could end up going above the max # of total idle or max total connection count:

- connection is removed
- in the interim (~100ms?) while detecting if the connection is alive, another mutex detects that it needs a connection and 1 more slot is available, so it creates one
- we add that back

Could get around that by doing this whole thing in a mutex, but it feels weird to lock a mutex while a call to an external dependency will be happening, could be a few hundred ms. 🤔